server/webapp/WEB-INF/rails/webpack/views/components/tab/index.tsx

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2019 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {bind} from "classnames/bind";
+import {MithrilComponent} from "jsx/mithril-component";
+import * as m from "mithril";
+import {Stream} from "mithril/stream";
+import * as stream from "mithril/stream";
+import * as styles from "./index.scss";
+
+const classnames = bind(styles);
+
+interface Attrs {
+  tabs: m.Child[] | any;
+  contents: m.Child[] | any;
+}
+
+interface State {
+  selectedTabIndex: Stream<number>;
+  setSelectedTabIndex: (index: number, e: MouseEvent) => void;
+  isSelected: (index: number) => boolean;
+}
+
+export class Tabs extends MithrilComponent<Attrs, State> {
+  oninit(vnode: m.Vnode<Attrs, State>): any {
+    vnode.state.selectedTabIndex = stream(0);
+
+    vnode.state.setSelectedTabIndex = (index: number, e: MouseEvent) => {
```

Comment:
Seems redundant, given that the stream can be used instead.

Reply:
The intention was to allow a user to navigate to component based on `#` of the tab. I have an HTML structure now to support it. I will update this in new PR.

Reply:
@ketan - Navigating to tab was a little tricky and positioning after scrolling to a position was getting complicated so I have dropped the idea.